rasa/utils/tensorflow/layers.py

```diff
@@ -589,8 +591,15 @@ def __init__(
         self.use_max_sim_neg = use_max_sim_neg
         self.neg_lambda = neg_lambda
         self.scale_loss = scale_loss
-        self.parallel_iterations = parallel_iterations
         self.same_sampling = same_sampling
+        self.constrain_similarities = constrain_similarities
+        self.model_confidence = model_confidence
+        self.similarity_type = similarity_type
+        if self.similarity_type and self.similarity_type not in {COSINE, INNER}:
+            raise RasaException(
```

Comment:
I don't think it should be `InvalidConfigException`. As vova had mentioned in the previous comment, this code can be used outside of regular Rasa usage where this TF layer is created without a usual rasa config. `RasaException` seems fine because of that.

Reply:
In that case I think you should still create a subclass for `RasaException` which is more specific, e.g. something like a `LayerConfigException`. `raise RasaException` is basically like a `raise Exception` and too broad

Reply:
Added `TFLayerConfigException`.